src/handle_init_contract.c

```diff
@@ -38,16 +38,13 @@ void handle_init_contract(ethPluginInitContract_t *msg) {
     }
 
     // Set `next_param` to be the first field we expect to parse.
-    // EDIT THIS: Adapt the `cases`, and set the `next_param` to be the first parameter you expect
-    // to parse.
     switch (context->selectorIndex) {
-        case SWAP_EXACT_ETH_FOR_TOKENS:
-            context->next_param = MIN_AMOUNT_RECEIVED;
+        case CLAIM_REGULAR_ACCOUNT:
```

Comment:
It would be better to use same case in one place, some thing like this:

```suggestion
    switch (context->selectorIndex) {
        case CLAIM_REGULAR_ACCOUNT:
        case CLAIM_MULTI_SIGNATURE_ACCOUNT:
            context->next_param = PROOF;
            break;
        default:
            PRINTF("Missing selectorIndex: %d\n", context->selectorIndex);
            msg->result = ETH_PLUGIN_RESULT_ERROR;
            return;
    }
```

Reply:
Applied!